cmd/pyroscope/command/agent.go

```diff
@@ -0,0 +1,54 @@
+package command
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pyroscope-io/pyroscope/pkg/cli"
+	"github.com/pyroscope-io/pyroscope/pkg/config"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func newAgentCmd(cfg *config.Agent) *cobra.Command {
+	agentCmd := &cobra.Command{
+		Use:   "agent [flags]",
+		Short: "starts pyroscope agent.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if cfg.Config != "" {
+				// Use config file from the flag.
+				viper.SetConfigFile(cfg.Config)
+
+				// If a config file is found, read it in.
+				if err := viper.ReadInConfig(); err == nil {
+					fmt.Fprintln(os.Stderr, "Using config file:", viper.ConfigFileUsed())
+				}
+
+				if err := viper.Unmarshal(cfg); err != nil {
+					fmt.Fprintln(os.Stderr, "Unable to unmarshal:", err)
+				}
+			}
```

Comment:
The error flow is most likely broken. And this fragment is very repetitive. Let it DRY:

```Golang
func loadConfig(path string, v interface{}) error {
	if path == "" {
		return nil
	}
	// I guess we should also handle case when the default config file is not found.
	viper.SetConfigFile(path)
	if err := viper.ReadInConfig(); err != nil {
		return err
	}
	return viper.Unmarshal(v)
}
```

Reply:
+1 on the DRYing

Reply:
ah yes, didn't notice it was repeating. 
As for 
`	if err := viper.ReadInConfig(); err != nil {
		return err
	}`
I am not sure what we are to do in case of error. As it will fallback to flags and env vars if they exist. Should we just log this error and continue? Or should we actually stop further execution if config file is not found or has an error?

Reply:
Good point. I'd advocate for fail fast strategy – if user specified a config file but it can not be read, it clearly says that this is a mistake/misconfiguration. Silent failures are pure evil.

Reply:
@Skemba at this point pyroscope fails to start if the default config file is not found. I'm wondering if it is intended.

Reply:
Just to clarify, it's a breaking change and we should be very careful.

if user specified a config file but it can not be read, it clearly says that this is a mistake/misconfiguration. Silent failures are pure evil.

Reply:
yeah it shouldn't fail if there's no config file